internal/js/modules/k6/browser/common/connection.go

```diff
@@ -327,18 +328,33 @@ func (c *Connection) findTargetIDForLog(id target.SessionID) target.ID {
 
 //nolint:funlen,gocognit,cyclop
 func (c *Connection) recvLoop() {
+	decoder := jlexer.Lexer{}
+	var buffer bytes.Buffer
 	c.logger.Debugf("Connection:recvLoop", "wsURL:%q", c.wsURL)
 	for {
-		_, buf, err := c.conn.ReadMessage()
+		_, reader, err := c.conn.NextReader()
 		if err != nil {
 			c.handleIOError(err)
 			return
 		}
 
+		if _, err := buffer.ReadFrom(reader); err != nil {
+			c.handleIOError(err)
+			return
+		}
+
+		// Clone the bytes from the recvBuffer to buf before unmarshaling
+		// the message. Without this, the unmarshaling will fail as it
+		// reuses the buf as the underlying bytes buffer.
+		buf := bytes.Clone(buffer.Bytes())
```

Comment:
@codebien thanks for the `sync.Pool` suggestion. I think there's some promise in going down that root. However, it's not going to be a straight forward change. Since the browser module works with events, it's not straight forward to keep track of all the events and have them put the buffer back in the pool. We'd need to refactor a lot of the codebase to be able to put the underlying buffer back in the pool, either by keeping track of the data or by allowing the handlers of the event to put the buffer back into the pool 😭